docs/quantize.md

```diff
@@ -0,0 +1,96 @@
+
+At this point there are two processes you can use. You can either use a Docker container to convert and quantize, OR you can manually run the scripts. The Docker container is the easiest way to do it, but it requires you to have Docker installed on your machine. If you don't have Docker installed, you can follow the manual process.
+
+## Convert and Quantize with Docker
+
+Run `docker run --rm -v /path/to/model/repo:/repo ollama/quantize -q quantlevel /repo`. For instance, if you have downloaded the latest Mistral 7B model, then clone it to your machine. Then change into that directory and you can run:
+
+```shell
+docker run --rm -v .:/repo ollama/quantize -q q4_0 /repo
+```
+
+You can find the different quantization levels below under **Quantize the Model**.
+
+This will output two files into the directory. First is a f16.bin file that is the model converted to GGUF. The second file is a q4_0.bin file which is the model quantized to a 4 bit quantization. You should rename it to something more descriptive.
+
+You can find the repository for the Docker container here: [https://github.com/mxyng/quantize](https://github.com/mxyng/quantize)
+
+## Convert and Quantize Manually
+
+### Clone llama.cpp to your machine
+
+If we know the model has a chance of working, then we need to convert and quantize. This is a matter of running two separate scripts in the llama.cpp project.
+
+1. Decide where you want the llama.cpp repository on your machine.
+2. Navigate to that location and then run:
+ [`git clone https://github.com/ggerganov/llama.cpp.git`](https://github.com/ggerganov/llama.cpp.git)
+    1. If you don't have git installed, download this zip file and unzip it to that location: https://github.com/ggerganov/llama.cpp/archive/refs/heads/master.zip
+3. Install the Python dependencies: `pip install torch transformers sentencepiece`
+
+### Convert the model to GGUF
+
+1. Decide on the right convert script to run. What was the model architecture you found in the first section.
+    1. LlamaForCausalLM or MistralForCausalLM:
+    run `python3 convert.py <modelfilename>`
+    No need to specify fp16 or fp32.
+    2. FalconForCausalLM or RWForCausalLM:
+    run `python3 convert-falcon-hf-to-gguf.py <modelfilename> <fpsize>`  
+    fpsize depends on the weight size. 1 for fp16, 0 for fp32
+    3. GPTNeoXForCausalLM:
+    run `python3 convert-gptneox-hf-to-gguf.py <modelfilename> <fpsize>`
+    fpsize depends on the weight size. 1 for fp16, 0 for fp32
+    4. GPTBigCodeForCausalLM:
+    run `python3 convert-starcoder-hf-to-gguf.py <modelfilename> <fpsize>`
+    fpsize depends on the weight size. 1 for fp16, 0 for fp32
+
+### Quantize the model
+
+If the model converted successfully, there is a good chance it will also quantize successfully. Now you need to decide on the quantization to use. We will always try to create all the quantizations and upload them to the library. You should decide which level is more important to you and quantize accordingly.
```

Comment:
This doesn't explain how to build the `quantize` binary. It's probably best to first start with a solid dock that uses the `ollama/quantize` helper container first, then we can add this next since it's much more involved